mllib/src/main/scala/org/apache/spark/ml/feature/Instance.scala

```diff
@@ -27,3 +27,25 @@ import org.apache.spark.ml.linalg.Vector
  * @param features The vector of features for this data point.
  */
 private[ml] case class Instance(label: Double, weight: Double, features: Vector)
+
+/**
+ * Case class that represents an instance of data point with
+ * label, weight, offset and features.
+ *
+ * @param label Label for this data point.
+ * @param weight The weight of this instance.
+ * @param offset The offset used for this data point.
+ * @param features The vector of features for this data point.
+ */
+private[ml] case class OffsetInstance(label: Double, weight: Double, offset: Double,
+                                   features: Vector) {
+
+  /** Constructs from an [[Instance]] object and offset */
+  def this(instance: Instance, offset: Double = 0.0) = {
+    this(instance.label, instance.weight, offset, instance.features)
+  }
+
+  /** Converts to an [[Instance]] object by leaving out the offset. */
+  private[ml] def toInstance: Instance = Instance(label, weight, features)
```

Comment:
it looks like this method is only used once in a test case, might it be better to remove it?